daphne_service_utils/src/durable_requests/mod.rs

```diff
@@ -0,0 +1,294 @@
+// Copyright (c) 2024 Cloudflare, Inc. All rights reserved.
+// SPDX-License-Identifier: BSD-3-Clause
+
+//! This module implements the communication between the daphne service and the storage proxy.
+//!
+//! Communication with the storage proxy happens through HTTP. The URI of the request is the URI
+//! to use when talking to the durable object, prefixed by [`DO_PATH_PREFIX`]. The body of this
+//! request will contain the metadata necessary for the storage proxy to locate the appropriate
+//! durable object, followed by the opaque body that should be forwarded to the durable object
+//! itself.
+//!
+//! # Metadata encoding
+//!
+//! The metadata is encoded using capnproto as the binary protocol, it contains:
+//! - The `binding` which identifies the Durable Object implementation to target.
+//! - The `id` which identifies the specific instance of the durable object.
+//!
+//! # Body encoding
+//!
+//! The body that follows this piece of metadata is opaque and to be understood by the Durable
+//! Object itself. So the proxy shall simply forward it along without any processing
+//!
+//! # The response
+//!
+//! The HTTP response is simply forwarded back from the Durable Object to the Native Service.
+//!
+//!```text
+//! +----------------+                 +---------------+            +----------------+
+//! | Native Service |                 | Storage Proxy |            | Durable Object |
+//! +----------------+                 +---------------+            +----------------+
+//!    | Http Request:                         |                            |
+//!    | POST /`DO_PATH_PREFIX`/path/to/call   |                            |
```

Comment:
Suggest versioning this in case we need to change something later on.
```suggestion
//!    | POST /v1/`DO_PATH_PREFIX`/path/to/call   |                            |
```